src/main/java/org/sonarlint/intellij/ui/SonarLintRulePanel.kt

```diff
@@ -102,69 +99,83 @@ class SonarLintRulePanel(private val project: Project) : JBLoadingPanel(BorderLa
         add(paramsPanel, BorderLayout.SOUTH)
 
         setLoadingText("Loading rule description...")
-        setRuleKey(null, null, null)
+        clear()
     }
 
 
-    fun setRuleKey(module: Module?, ruleKey: String?, contextKey: String?) {
-        if (currentModule == module && currentRuleKey == ruleKey && currentContextKey == contextKey) {
-            return
-        }
-        currentRuleKey = ruleKey
-        currentModule = module
-        currentContextKey = contextKey
-        if (module == null || ruleKey == null) {
-            nothingToDisplay(false)
-            return
+    inner class SmartRuleDetailsLoader {
```

Comment:
Just removing Smart would work for me. But no strong pushback from me, I think it's not that important.
You could also group the 3 `last` fields in one and use a Kotlin data class, that would ease comparison.

Reply:
Good idea to use a data class!